common/cmake/gen.cmake

```diff
@@ -115,15 +114,85 @@ function(V2X)
       ${INCLUDE_ARG}
     WORKING_DIRECTORY ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/
   )
-
   add_file_target(FILE "${V2X_NAME}.model.xml" GENERATED)
   get_file_target(SRC_TARGET_NAME "${V2X_NAME}.model.xml")
   set_target_properties(${SRC_TARGET_NAME} PROPERTIES INCLUDE_FILES "${MODEL_INCLUDE_FILES}")
 
+  add_custom_command(
+    OUTPUT "${V2X_NAME}.arch.xml"
+    DEPENDS
+      ${DEPENDS_LIST}
+      ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/vlog_to_test_arch.py
+      ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/template.arch.xml
+    COMMAND
+      ${CMAKE_COMMAND} -E env YOSYS=${YOSYS}  ${PYTHON3} ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/vlog_to_test_arch.py ${TOP_ARG}
+      -o ${CMAKE_CURRENT_BINARY_DIR}/${V2X_NAME}.arch.xml ${FIRST_SOURCE}
+      ${INCLUDE_ARG}
+    WORKING_DIRECTORY ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/
+  )
+  add_file_target(FILE "${V2X_NAME}.arch.xml" GENERATED)
+
+  add_custom_command(
+    OUTPUT "${V2X_NAME}.test.eblif"
+    DEPENDS
+      ${DEPENDS_LIST}
+      ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/vlog_to_test_eblif.py
+    COMMAND
+      ${CMAKE_COMMAND} -E env YOSYS=${YOSYS}  ${PYTHON3} ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/vlog_to_test_eblif.py ${TOP_ARG}
+      -o ${CMAKE_CURRENT_BINARY_DIR}/${V2X_NAME}.test.eblif ${FIRST_SOURCE}
+      ${INCLUDE_ARG}
+    WORKING_DIRECTORY ${symbiflow-arch-defs_SOURCE_DIR}/utils/vlog/
+  )
+  add_file_target(FILE "${V2X_NAME}.test.eblif" GENERATED)
+
+  xml_sort(
+    NAME ${V2X_NAME}_arch_merged
+    FILE ${V2X_NAME}.arch.xml
+    OUTPUT ${V2X_NAME}.arch.merged.xml
+  )
+
+  get_target_property_required(VPR env VPR)
+  get_target_property(VPR_TARGET env VPR_TARGET)
+
+  add_custom_command(
+    OUTPUT
+      ${V2X_NAME}.vpr.stdout
+    DEPENDS
+      ${V2X_NAME}.arch.merged.xml
+      ${V2X_NAME}.test.eblif
+      ${QUIET_CMD} ${QUIET_CMD_TARGET}
+      ${VPR} ${VPR_TARGET}
+    COMMAND
+      ${QUIET_CMD} ${VPR} ${V2X_NAME}.arch.merged.xml
+      ${V2X_NAME}.test.eblif
+      --route_chan_width 100
+      --min_route_chan_width_hint 1
+      --outfile_prefix ${V2X_NAME}
+      --pack
+      --place
+      --route
+    COMMAND
+      ${CMAKE_COMMAND} -E copy vpr_stdout.log
```

Comment:
This is unsafe to execute in parallel, put this in a subdirectory, or make sure VPR writes to a unique file name.

Reply:
How does one do this? I just copied it from the `devices.cmake`

Reply:
devices.cmake does execute VPR in seperate working directories, which solves the parallel execution problem.

Reply:
```
          //Allow the default vpr log file to be overwritten                     
          const char* env_value = std::getenv("VPR_LOG_FILE");   
```

So set "VPR_LOG_FILE" to something else

Reply:
I put them into separate directory.